src/main/java/it/gov/pagopa/paymentupdater/service/PaymentServiceImpl.java

```diff
@@ -70,52 +78,63 @@ public Optional<Payment> getPaymentByNoticeNumberAndFiscalCode(String noticeNumb
 	public void save(Payment reminder) {
 		paymentRepository.save(reminder);
 		log.info("Saved payment id: {}", reminder.getId());
-	}
+	} 
 
 	@Override
-	public Map<String, Boolean> checkPayment(Payment payment)
-			throws JsonProcessingException, InterruptedException, ExecutionException {
-		Map<String, Boolean> map = new HashMap<>();
-		map.put(isPaid, false);
-		try {
+	public Map<String, String> checkPayment(Payment payment) throws JsonProcessingException, InterruptedException, ExecutionException {
+		Map<String, String> map = new HashMap<>();
+		map.put(isPaid, "false");
+		try {			
 			ApiClient apiClient = new ApiClient();
 			if (enableRestKey) {
-				apiClient.addDefaultHeader("Ocp-Apim-Subscription-Key", proxyEndpointKey);
+				apiClient.setApiKey(proxyEndpointKey);
```

Comment:
we must set an `Ocp-Apim-Subscription-Key` because Proxy Apis are exposed by APIM that expect this header, that is not declared in specs

Reply:
fixed